File: problems/409-longest-palindrome/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        counts = {}
        for char in s:
            counts[char] = counts.get(char, 0) + 1
        
        length = 0
        odd_exists = False
        for count in counts.values():
            if count % 2 == 0:
                length += count
            else:
                length += count - 1
                odd_exists = True
        
        if odd_exists:
            length += 1
        return length

File: problems/409-longest-palindrome/solution.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> counts;
        for (char c : s) {
            counts[c]++;
        }
        int length = 0;
        int oddCount = 0;
        for (auto const& [key, val] : counts) {
            if (val % 2 == 0) {
                length += val;
            } else {
                length += val - 1;
                oddCount = 1;
            }
        }
        return length + oddCount;
    }
};